src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -950,7 +950,7 @@ public static sbyte Max(sbyte val1, sbyte val2)
         public static float Max(float val1, float val2)
         {
             if (AdvSimd.Arm64.IsSupported &&
-                (!RuntimeHelpers.IsKnownConstant(val1) || !RuntimeHelpers.IsKnownConstant(val1)))
+                (!RuntimeHelpers.IsKnownConstant(val1) || !RuntimeHelpers.IsKnownConstant(val2)))
```

Comment:
You shouldn't need `IsKnownConstant` on Arm64. The underlying instruction is already IEEE 754 compliant.

I'd also recommend doing this in import in the JIT rather than in C# to improve JIT throughput and allow us to better optimize it (CSE, value numbering, constant folding, etc).

Reply:
> You shouldn't need IsKnownConstant on Arm64. 

it's needed because SIMD APIs don't do constant folding - I didn't want to regress constant input case

Reply:
You could just update CSE to do the folding here so that we could get it; or have it handled like `Round`, `Ceil`, `Floor`, `Truncate`, etc where its carried through up until lowering as `NI_Math_Max` and then lowered to the correct IR.

Reply:
I was too lazy to do all of that to find out it doesn't pass some edge-case test 🙂 let's wait for CI to finish

Reply:
@tannergooding also, I don't think this can go through VN, CSE - it's a fully managed AggressiveInlining method, it's expanded during import, so handling it in JIT doesn't add much value other than throughput

Reply:
Tests passed!